### PR TITLE
feat: use vite server with hmr for development

### PR DESCRIPTION
**Pull request by triggertoo**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,7 +37,8 @@ Params:
 | ---------- | ----------- | ---- | ------------- |
 | contentBlockPath | Directory of the block | string | Current dir |
 | entryPath | Path to the entry file exporting `block` and `settings` | string | `src/index.tsx` |
-| port | Port for the local web server | string | 5600 |
+| port | Port for the local development server | number | 5600 |
+| metaPort | Port for the local meta information (url, entrypoint, port) | number | 5601 |
```
Comment (SamuelAlev) [inline README.md:41]: You can probably register the meta endpoint on the vite server

Comment (triggertoo) [reply README.md]: using middleware mode now: https://github.com/Frontify/frontify-cli/pull/232/commits/a67394644891ab94952be5f58b807d7ce16c4d3c

File: src/utils/compiler.ts [hunk 2]
```diff
@@ -24,15 +25,9 @@ export const getViteConfig = (
...
-
-            watch: watch
-                ? {
-                      include: `${projectPath}/**`,
-                      exclude: filesToIgnore,
-                  }
-                : null,
             rollupOptions: {
                 external: ['react', 'react-dom'],
+                plugins: [],
```
Comment (SamuelAlev) [inline src/utils/compiler.ts:30]:
```suggestion
```
No needed right?

File: package.json [hunk 3]
```diff
@@ -25,43 +25,35 @@
...
         "@fastify/cors": "^8.0.0",
-        "@fastify/static": "^6.4.0",
-        "@fastify/websocket": "^6.0.1",
+        "@vitejs/plugin-react": "^2.0.0",
         "fastify": "4.2.1",
         "minimist": "1.2.6",
         "minimist-options": "4.1.0",
-        "vite": "2.9.14"
+        "vite": "3.0.3",
+        "vite-plugin-environment": "^1.1.2",
```
Comment (SamuelAlev) [inline package.json:33]:
```suggestion
```
Doesn't seems to be used

File: src/utils/compiler.test.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+declare global {
+    interface Window {
+        index: { block: unknown; settings: unknown };
+    }
+}
+
+global.window = {} as any;
+
+describe('Compiler utils', async () => {
+    test('should provide a valid build with block and settings', async () => {
```
Comment (SamuelAlev) [inline src/utils/compiler.test.ts:19]: Here are few more tests cases:
- Ensure `react` & `react-dom` are loaded from `window`
- Ensure `process.env.NODE_ENV` is correctly replaced
- Ensure when `process.env.NODE_ENV = 'production'` that code with `process.env.NODE_ENV === 'development'` is correctly removed (dead code elimination)

File: src/utils/compiler.test.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+const outputFile = `${__dirname}/../../dist/index.js`;
+const pathToIndex = '__mocks__/index.tsx';
+
+declare global {
+    interface Window {
+        index: { block: unknown; settings: unknown };
+    }
+}
+
+global.window = {} as any;
```
Comment (SamuelAlev) [inline src/utils/compiler.test.ts:16]: Should be in a `beforeAll` no?

File: src/utils/compiler.test.ts (hunk 4 above)
Comment (triggertoo) [reply src/utils/compiler.test.ts:19]: ah definitely, I just pushed the first one 😉 more to come

**Review by SamuelAlev — APPROVED** [review]

**Pull request merged**